Attempt to define bidirectional association for external class causes exception

## Example

```markdown
There is a WebApp.
There is a Page.
WebApp has foo and is bar of Page.
```

This does not work in the web app, because it does not run the compiler with `--imports org.fulib.mockups`.

## Output

```
failed to execute phase 'NameResolver' due to exception:
java.lang.NullPointerException
	at org.fulib.scenarios.visitor.resolve.DeclResolver.resolveAssociation(DeclResolver.java:347)
	at org.fulib.scenarios.visitor.resolve.SentenceResolver.resolveHasNamedExpr(SentenceResolver.java:411)
	at org.fulib.scenarios.visitor.resolve.SentenceResolver.visit(SentenceResolver.java:332)
	at org.fulib.scenarios.visitor.resolve.SentenceResolver.visit(SentenceResolver.java:38)
	at org.fulib.scenarios.ast.sentence.HasSentence.accept(HasSentence.java:24)
	at org.fulib.scenarios.visitor.resolve.SentenceResolver.visit(SentenceResolver.java:54)
	at org.fulib.scenarios.visitor.resolve.SentenceResolver.visit(SentenceResolver.java:38)
	at org.fulib.scenarios.ast.sentence.SentenceList.accept(SentenceList.java:20)
	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:138)
	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:20)
	at org.fulib.scenarios.ast.Scenario.accept(Scenario.java:24)
	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:106)
	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:20)
	at org.fulib.scenarios.ast.ScenarioFile.accept(ScenarioFile.java:30)
	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:59)
	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:20)
	at org.fulib.scenarios.ast.ScenarioGroup.accept(ScenarioGroup.java:26)
	at org.fulib.scenarios.visitor.resolve.NameResolver.lambda$visit$0(NameResolver.java:44)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734)
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)
	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:44)
	at org.fulib.scenarios.ast.CompilationContext.accept(CompilationContext.java:24)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:145)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:129)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:97)
	at org.fulib.scenarios.Main.main(Main.java:9)
failed to execute phase 'CodeGenerator' due to exception:
java.lang.NullPointerException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
	at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:735)
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:51)
	at org.fulib.scenarios.ast.CompilationContext.accept(CompilationContext.java:24)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:145)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:129)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:97)
	at org.fulib.scenarios.Main.main(Main.java:9)
Caused by: java.lang.NullPointerException
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:307)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:31)
	at org.fulib.scenarios.ast.ScenarioFile.accept(ScenarioFile.java:30)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.populateTests(CodeGenerator.java:163)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:148)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:31)
	at org.fulib.scenarios.ast.ScenarioGroup.accept(ScenarioGroup.java:26)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.lambda$visit$0(CodeGenerator.java:55)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3566)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

--- a/src/main/java/org/fulib/scenarios/visitor/resolve/DeclResolver.java
+++ b/src/main/java/org/fulib/scenarios/visitor/resolve/DeclResolver.java
@@ -328,6 +328,11 @@ public class DeclResolver
       }
 
       final AssociationDecl association = createAssociation(scope, position, owner, name, cardinality, otherClass);
+      if (association == null)
+      {
+         // class was external or frozen, error already reported
+         return null;
+      }
 
       if (otherClass == owner && name.equals(otherName))
       {
@@ -343,9 +348,11 @@ public class DeclResolver
       {
          final AssociationDecl other = createAssociation(scope, otherPosition, otherClass, otherName,
                                                          otherCardinality, owner);
-
-         association.setOther(other);
-         other.setOther(association);
+         if (other != null)
+         {
+            association.setOther(other);
+            other.setOther(association);
+         }
       }
 
       return association;

--- a/src/test/invalid_scenarios/lang/Assocs.md
+++ b/src/test/invalid_scenarios/lang/Assocs.md
@@ -84,3 +84,18 @@ error: conflicting redeclaration of 'AssocReEntity.related' [property.redeclarat
 was: association to one 'AssocReEntity'
 now: association to one 'AssocReEntityOther'
 -->
+
+## External Associations
+
+There is a WebApp.
+There is a Page.
+WebApp has foo and is bar of Page.
+<!--       ^
+error: cannot resolve or add association 'foo' in external class 'WebApp' [association.unresolved.external]
+-->
+
+There is a NonExternal.
+NonExternal has webapp and is nonExternal of WebApp.
+<!--                          ^
+error: cannot resolve or add association 'nonExternal' in external class 'WebApp' [association.unresolved.external]
+-->

--- a/src/test/invalid_scenarios/lang/Attributes.md
+++ b/src/test/invalid_scenarios/lang/Attributes.md
@@ -125,3 +125,11 @@ note: perhaps you meant to refer to 'erbi' instead of the string literal 'Herbi'
      ^
 note: perhaps you meant to refer to 'herba' instead of the string literal 'Herbi'? [stringliteral.typo]
 -->
+
+## External Attributes
+
+There is a WebApp.
+WebApp has 1 foo.
+<!--         ^
+error: cannot resolve or add attribute 'foo' in external class 'WebApp' [attribute.unresolved.external]
+-->

